src/engine/Operation.cpp

```diff
@@ -64,7 +66,7 @@ void Operation::recursivelySetCancellationHandle(
 void Operation::recursivelySetTimeConstraint(
     std::chrono::steady_clock::time_point deadline) {
   deadline_ = deadline;
-  forAllDescendants([deadline](auto child) {
+  forAllDescendantsImpl([deadline](auto child) {
```

Comment:
Revert all the changes that are no longer relevant to reduce merge conflicts with other people.